src/CouchDB.Driver/ICouchDatabase.cs

```diff
@@ -88,6 +88,31 @@ public interface ICouchDatabase<TSource>: IOrderedQueryable<TSource>
         /// <returns>A task that represents the asynchronous operation. The task result contains the elements created or updated.</returns>
         Task<IEnumerable<TSource>> AddOrUpdateRangeAsync(IList<TSource> documents, CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Execute a couchdb view and get a result with the values.
+        /// </summary>
+        /// <typeparam name="TValue">The type of the value that will be returned.</typeparam>
+        /// <param name="design">The design to use.</param>
+        /// <param name="view">The view to use.</param>
+        /// <param name="options">Optional options to pass to the view.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> to observe while waiting for the task to complete.</param>
+        /// <returns>A task that represents the asynchronous operation. The task result contains the <see cref="CouchViewResult{TValue}"/>.</returns>
+        Task<CouchViewResult<TValue>> GetViewAsync<TValue>(string design, string view, CouchViewOptions? options = null, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Execute a couchdb view and get a result with the values and the docs.
+        /// </summary>
+        /// <typeparam name="TValue">The type of the value that will be returned.</typeparam>
+        /// <typeparam name="TDoc">The type of the document that will be returned.</typeparam>
+        /// <param name="design">The design to use.</param>
+        /// <param name="view">The view to use.</param>
+        /// <param name="options">Optional options to pass to the view.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> to observe while waiting for the task to complete.</param>
+        /// <returns>A task that represents the asynchronous operation. The task result contains the <see cref="CouchViewResult{TValue, TDoc}"/>.</returns>
+        /// <remarks>The options IncludeDocs will always be set to true.</remarks>
+        Task<CouchViewResult<TValue, TDoc>> GetViewAsync<TValue, TDoc>(string design, string view, CouchViewOptions? options = null, CancellationToken cancellationToken = default)
```

Comment:
Should it be `TSource` instead of `TDoc`?

Reply:
I was thinking about this too. I was using multiple views that had different docs that's why I used a `TDoc`. Also, it might be better this way if the docs have a different schema? If you believe it's better to use `TSource` I have no problem changing it this way since it will align with how the library works.

Reply:
Yeah `TSource` is more about consistency